Userland/Libraries/LibJS/Runtime/Intl/CollatorCompareFunction.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, Idan Horowitz <idan.horowitz@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Utf16View.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/Intl/Collator.h>
+#include <LibJS/Runtime/Intl/CollatorCompareFunction.h>
+
+namespace JS::Intl {
+
+CollatorCompareFunction* CollatorCompareFunction::create(GlobalObject& global_object, Collator& collator)
+{
+    return global_object.heap().allocate<CollatorCompareFunction>(global_object, global_object, collator);
+}
+
+CollatorCompareFunction::CollatorCompareFunction(GlobalObject& global_object, Collator& collator)
+    : NativeFunction(*global_object.function_prototype())
+    , m_collator(collator)
+{
+}
+
+void CollatorCompareFunction::initialize(GlobalObject& global_object)
+{
+    auto& vm = global_object.vm();
+    define_direct_property(vm.names.length, Value(2), Attribute::Configurable);
+}
+
+double compare_strings(Collator& collator, Utf16View const& x, Utf16View const& y)
```

Comment:
I know the whole thing is implementation defined, but we can probably have the spec link still: https://tc39.es/ecma402/#sec-collator-comparestrings

Reply:
Sure, added